### PR TITLE
updating learning sidebar structure for new web forms module

### PR DESCRIPTION
**Pull request by chrisdavidmills**
I've published the new web forms module on the learning area — https://wiki.developer.mozilla.org/en-US/docs/Learn/Forms.

This is a significant update to the HTML forms module that adds lots of new examples and better descriptions, but also reorganizes the articles and moves it out from the HTML topic, as it involves a lot more than just HTML.

This PR aims to adjust the sidebar to account for this new structure.

I've not tested it yet on my local Kuma; this will be the next step.

### CODE REVIEW

**Comment by chrisdavidmills** [comment]
Update on testing — I can't do this yet because I've not got wiki.dmo set up on my local install of Kuma. Will liaise with @escattone soon to try and figure this out.

**Comment by escattone** [comment]
@chrisdavidmills First, here's some setup you should do:

1. Get the latest `kuma` code (`git checkout master` --> `git pull --ff-only mdn master`), and go to the root directory of your local `kuma` repo.
2. Copy `.env-dist.dev` to `.env`
3. Update your `/etc/hosts` file to:
   ```
   127.0.0.1       localhost demos localhost.org wiki.localhost.org
   255.255.255.255 broadcasthost
   ::1             localhost demos localhost.org wiki.localhost.org
   ```
1. `docker-compose down` (if not already)
1. `docker-compose pull` (to get the latest Docker images)
1. Get the **your** `kumascript` code (`cd kumascript` --> `git checkout learn-sidebar-webforms`)
1. `docker-compose up -d` (to spin everything up in the background)
1. `docker-compose exec web ./manage.py migrate` (to update your local DB)
1. `docker-compose ps` (to check that all of the services area running -- although the `minio` setup task should be in the `exited` state)
1. Login to the admin console (http://localhost.org:8000/admin) using the `test-super` user with a password of `test-password` and update the `Sites` section: click on `Change` --> click on the entry that says `localhost:8000` --> change both the domain and display name from `localhost:8000` to `localhost.org:8000` --> click `Save`.
1. It may be sufficient to just render one document that you'd like to check, for example `docker-compose exec web ./manage.py render_document /en-US/docs/Learn/Forms` (that will be very fast), or you may want to re-render all of the documents in your sample DB, `docker-compose exec web ./manage.py render_document --all` (that would take several minutes).
1. Then go to `http://wiki.localhost.org:8000/en-US/docs/Learn/Forms` (you can do shift-reloads to kick-off re-renders) or you can go to `http://localhost.org:8000/en-US/docs/Learn/Forms` (but you won't be able to manually start re-renders by doing a shift-reload).

**Comment by chrisdavidmills** [comment]
@escattone you are a genius sir! I followed your docker/django voodoo, and it worked like a charm!

So, I've tested mymacro update locally, including scraping all the form pages linked to by the new sidebar. The links all work fine, and I just fixed a typo in one of the link labels (`from controls` -> `form controls`).

I think this is probably read to merge, but maybe someone else could try running it?

**Comment by escattone** [comment]
@chrisdavidmills that's so awesome to hear! it's so nice when things work the first time! i'm going to save your comment, and read it to myself on those other days when things don't work the first time! 😄

**Comment by chrisdavidmills** [comment]
@Elchi3 please can you review this one for me? The learn sidebar is now out of date, and we are getting bugs filed about it ;-)

File: macros/LearnSidebar.ejs [hunk 1]
```diff
@@ -152,6 +139,23 @@ var text = mdn.localStringMap({
         'Client-side_web_APIs_Drawing_graphics': 'Drawing graphics',
         'Client-side_web_APIs_Video_and_audio_APIs': 'Video and audio APIs',
         'Client-side_web_APIs_client-side_storage': 'Client-side storage',
+    'Web_forms' : 'Web forms — Collecting data from users',
```
Comment (Elchi3) [inline macros/LearnSidebar.ejs:142]: I wouldn't name this "Collecting data from users" given how data collection is quite negatively connoted these days. How about "Web forms – Working with user data" ? Just a thought, though.

File: macros/LearnSidebar.ejs [hunk 2]
```diff
@@ -152,6 +139,23 @@ var text = mdn.localStringMap({
...
+        'Other_form_controls': 'Other form controls',
+        'Styling_web_forms' : 'Styling web forms',
+        'Advanced_form_styling' : 'Advanced form styling',
+        'UI_pseudo-classes': 'UI pseudo-classes',
+        'Client-side_form_validation' : 'Client-side form validation',
+        'Sending_form_data' : 'Sending form data',
+      'Web_forms_advanced': 'Advanced forms articles',
+        'How_to_build_custom_form_controls' : 'How to build custom form controls',
+        'Sending_forms_through_JavaScript' : 'Sending forms through JavaScript',
+        'CSS_Property_compatibility_table_for_form_controls' : 'CSS Property compatibility table for form controls',
```
Comment (Elchi3) [inline macros/LearnSidebar.ejs:158]: Please apply this throughout.

```suggestion
        'CSS_Property_compatibility_table_for_form_controls' : 'CSS property compatibility table for form controls',
```

**Comment by chrisdavidmills** [comment]
Thanks @Elchi3 , that was really good feedback! Updates made as requested, and I've changed the page titles on MDN accordingly.

**Pull request merged**